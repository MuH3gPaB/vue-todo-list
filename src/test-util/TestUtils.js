import {jest} from "@jest/globals";

export const mockModule = function (module, childModules) {
    let mockedModule = {actions: {}, getters: {}, state: {}, modules: childModules};
    Object.keys(module.actions || []).forEach(key => mockedModule.actions[key] = jest.fn());
    Object.keys(module.getters || []).forEach(key => mockedModule.getters[key] = jest.fn());
    Object.keys(module.state || []).forEach(key => mockedModule.state[key] = module.state[key]);
    return mockedModule;
}
