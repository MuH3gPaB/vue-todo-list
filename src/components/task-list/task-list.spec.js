import {beforeEach, describe, it} from "@jest/globals";
import TaskList from './task-list';
import ToDoListModule from '@/store/to-do-list';
import {createLocalVue, mount} from "vue-test-utils";
import Vuex from 'vuex';
import {mockModule} from "@/test-util/TestUtils";
import TaskModule from "@/store/task";

function mockStore(tasks) {
    let tasksModules = tasks.reduce((result, task, idx) => {
        return {
            ...result,
            ['Task' + idx]: mockModule(TaskModule(task))
        }
    }, {});

    return mockModule({}, {
        ToDoListModule: mockModule(ToDoListModule, tasksModules)
    });
}

describe('TaskList', () => {
    let localVue;
    let store;
    let wrapper;

    beforeEach(() => {
        localVue = createLocalVue();
        localVue.use(Vuex);
        store = mockStore(testTasks());
        store.modules.ToDoListModule.getters.getTasksSortOptions.mockReturnValue(testSorts());
        store.modules.ToDoListModule.getters.getOpenTaskModuleIds.mockReturnValue(['Task0', 'Task1']);
        store.modules.ToDoListModule.getters.getDoneTaskModuleIds.mockReturnValue(['Task2', 'Task3']);

        wrapper = mount(TaskList, {
            localVue,
            store: new Vuex.Store(store)
        });
    });

    it('should match snapshot', () => {
        expect(wrapper.vm.$el).toMatchSnapshot();
    });
});

function testTasks() {
    return [
        {
            id: 0,
            text: 'Test Task1',
            isDone: false,
            createdDate: '2020-10-15T19:24:52.829Z',
            dueDate: null
        },
        {
            id: 1,
            text: 'Test Task2',
            isDone: false,
            createdDate: '2020-10-15T19:24:52.829Z',
            dueDate: null
        },
        {
            id: 2,
            text: 'Test Task3',
            isDone: true,
            createdDate: '2020-10-15T19:24:52.829Z',
            dueDate: '2020-10-15T19:54:52.829Z'
        },
        {
            id: 3,
            text: 'Test Task4',
            isDone: true,
            createdDate: '2020-10-15T19:24:52.829Z',
            dueDate: '2020-10-15T19:54:52.829Z'
        },
    ];
}

function testSorts() {
    return {
        CREATED_AT_ASC: {
            label: 'Date creation (Asc)',
            comparator: (t1, t2) => (t1.createdDate > t2.createdDate ? 1 : -1)
        },
        CREATED_AT_DESC: {
            label: 'Date creation (Desc)',
            comparator: (t1, t2) => (t1.createdDate < t2.createdDate ? 1 : -1)
        },
        DUE_DATE_ASC: {
            label: 'Due date (Asc)',
            comparator: (t1, t2) => (t1.dueDate > t2.dueDate ? 1 : -1)
        },
        DUE_DATE_DESC: {
            label: 'Due date (Desc)',
            comparator: (t1, t2) => (t1.dueDate < t2.dueDate ? 1 : -1)
        },
        TEXT_ASC: {
            label: 'Text (Asc)',
            comparator: (t1, t2) => (t1.text > t2.text ? 1 : -1)
        },
        TEXT_DESC: {
            label: 'Text (Desc)',
            comparator: (t1, t2) => (t1.text < t2.text ? 1 : -1)
        }
    };
}
