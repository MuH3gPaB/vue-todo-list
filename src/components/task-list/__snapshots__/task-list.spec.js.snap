// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList should match snapshot 1`] = `
<div
  class="task-list"
>
  <div
    class="task-list-container"
  >
    <div>
      <div
        class="todo-list-header form-inline"
      >
        <label
          class="todo-list-label"
          for="search-input"
        >
          Todo List
        </label>
         
        <input
          class="input-lg form-control search-input"
          id="search-input"
          placeholder="Search..."
          type="text"
        />
      </div>
    </div>
     
    <form
      class="todo-list-add form-inline"
    >
      <input
        class="new-task-input form-control input-lg"
        formaction="#"
        placeholder="New task..."
        type="text"
      />
       
      <input
        class="btn-lg add-button"
        type="button"
        value="ADD"
      />
    </form>
     
    <div
      class="task-bar-list"
    >
      <div
        class="bar-list-header"
      >
        <p
          class="bar-list-label"
        >
          Open
        </p>
         
        <select
          class="form-control input-lg sort-select"
        >
          <option
            value="CREATED_AT_ASC"
          >
            
        Date creation (Asc)
      
          </option>
          <option
            value="CREATED_AT_DESC"
          >
            
        Date creation (Desc)
      
          </option>
          <option
            value="TEXT_ASC"
          >
            
        Text (Asc)
      
          </option>
          <option
            value="TEXT_DESC"
          >
            
        Text (Desc)
      
          </option>
        </select>
      </div>
       
      <div
        class="bar-list-body"
      >
        <div
          class="task-bar"
        >
          <div
            class="task"
          >
            <input
              class="form-control checkbox"
              type="checkbox"
            />
             
            <p
              class="text"
            >
              
      Test Task1
    
            </p>
             
            <textarea
              class="text text-input"
              style="display: none;"
            />
             
            <div
              class="time"
            >
              <p
                class="created"
              >
                10:24 PM
              </p>
               
              <p
                class="completed"
              >
                
              </p>
            </div>
          </div>
           
          <div
            class="delete"
            style="display: none;"
          />
        </div>
        <div
          class="task-bar"
        >
          <div
            class="task"
          >
            <input
              class="form-control checkbox"
              type="checkbox"
            />
             
            <p
              class="text"
            >
              
      Test Task2
    
            </p>
             
            <textarea
              class="text text-input"
              style="display: none;"
            />
             
            <div
              class="time"
            >
              <p
                class="created"
              >
                10:24 PM
              </p>
               
              <p
                class="completed"
              >
                
              </p>
            </div>
          </div>
           
          <div
            class="delete"
            style="display: none;"
          />
        </div>
      </div>
       
      <div
        class="bar-list-footer"
      >
        <a
          class="bar-list-clear"
          href="#"
        >
          
      Clear "Open" list
    
        </a>
      </div>
    </div>
     
    <div
      class="task-bar-list"
    >
      <div
        class="bar-list-header"
      >
        <p
          class="bar-list-label"
        >
          Done
        </p>
         
        <select
          class="form-control input-lg sort-select"
        >
          <option
            value="DUE_DATE_ASC"
          >
            
        Due date (Asc)
      
          </option>
          <option
            value="DUE_DATE_DESC"
          >
            
        Due date (Desc)
      
          </option>
          <option
            value="TEXT_ASC"
          >
            
        Text (Asc)
      
          </option>
          <option
            value="TEXT_DESC"
          >
            
        Text (Desc)
      
          </option>
        </select>
      </div>
       
      <div
        class="bar-list-body"
      >
        <div
          class="task-bar"
        >
          <div
            class="task"
          >
            <input
              class="form-control checkbox"
              type="checkbox"
            />
             
            <p
              class="text"
            >
              
      Test Task3
    
            </p>
             
            <textarea
              class="text text-input"
              style="display: none;"
            />
             
            <div
              class="time"
            >
              <p
                class="created"
              >
                10:24 PM
              </p>
               
              <p
                class="completed"
              >
                10:54 PM
              </p>
            </div>
          </div>
           
          <div
            class="delete"
            style="display: none;"
          />
        </div>
        <div
          class="task-bar"
        >
          <div
            class="task"
          >
            <input
              class="form-control checkbox"
              type="checkbox"
            />
             
            <p
              class="text"
            >
              
      Test Task4
    
            </p>
             
            <textarea
              class="text text-input"
              style="display: none;"
            />
             
            <div
              class="time"
            >
              <p
                class="created"
              >
                10:24 PM
              </p>
               
              <p
                class="completed"
              >
                10:54 PM
              </p>
            </div>
          </div>
           
          <div
            class="delete"
            style="display: none;"
          />
        </div>
      </div>
       
      <div
        class="bar-list-footer"
      >
        <a
          class="bar-list-clear"
          href="#"
        >
          
      Clear "Done" list
    
        </a>
      </div>
    </div>
  </div>
</div>
`;
